assets/js/modules/adsense/components/settings/AdBlockingRecoveryApp/index.js

```diff
@@ -0,0 +1,223 @@
+
+export default function AdBlockingRecoveryApp() {
+	const breakpoint = useBreakpoint();
+
+	const settingsURL = useSelect(
+		( select ) =>
+			`${ select( CORE_SITE ).getAdminURL(
+				'googlesitekit-settings'
+			) }#/connected-services/adsense`
+	);
+	const initialActiveStep = useSelect( ( select ) => {
+		const adBlockingRecoverySetupStatus =
+			select( MODULES_ADSENSE ).getAdBlockingRecoverySetupStatus();
+
+		switch ( adBlockingRecoverySetupStatus ) {
+			case '':
+				return 0;
+			case AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED:
+				return 1;
+			case AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED:
+				return 2;
+		}
+	} );
+	const createMessageCTAClicked = useSelect(
+		( select ) =>
+			!! select( CORE_UI ).getValue(
+				AD_BLOCKING_RECOVERY_SETUP_CREATE_MESSAGE_CTA_CLICKED
+			)
+	);
+	const dashboardURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard' )
+	);
+
+	const {
+		saveSettings,
+		setAdBlockingRecoverySetupStatus,
+		setUseAdBlockerDetectionSnippet,
+		setUseAdBlockerDetectionErrorSnippet,
+	} = useDispatch( MODULES_ADSENSE );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+
+	const [ activeStep, setActiveStep ] = useState( initialActiveStep );
+
+	const onCancel = useCallback( async () => {
+		if ( activeStep === 0 ) {
+			return navigateTo( dashboardURL );
+		}
+
+		if ( createMessageCTAClicked ) {
+			return navigateTo( settingsURL );
+		}
+
+		setAdBlockingRecoverySetupStatus( '' );
+		setUseAdBlockerDetectionSnippet( false );
+		setUseAdBlockerDetectionErrorSnippet( false );
+
+		const { error } = await saveSettings();
+
+		if ( ! error ) {
+			navigateTo( dashboardURL );
+		}
+	}, [
+		activeStep,
+		createMessageCTAClicked,
+		dashboardURL,
+		navigateTo,
+		saveSettings,
+		setAdBlockingRecoverySetupStatus,
+		setUseAdBlockerDetectionErrorSnippet,
+		setUseAdBlockerDetectionSnippet,
+		settingsURL,
+	] );
+
+	useEffect( () => {
+		if ( undefined === activeStep && undefined !== initialActiveStep ) {
+			setActiveStep( initialActiveStep );
+		}
+	}, [ activeStep, initialActiveStep ] );
+
+	const isTabletWidthOrLarger = breakpoint !== BREAKPOINT_SMALL;
+
+	return (
+		<Fragment>
+			<Header>
+				<HelpMenu />
+			</Header>
+			<div className="googlesitekit-ad-blocking-recovery googlesitekit-module-page">
+				<Grid>
+					<Row>
+						<Cell size={ 12 }>
+							<Layout rounded>
+								<Grid>
+									<Row>
+										<Cell
+											lgSize={ 6 }
+											mdSize={ 8 }
+											smSize={ 4 }
+										>
+											<PageHeader
+												className="googlesitekit-heading-3 googlesitekit-ad-blocking-recovery__heading"
+												title={ __(
+													'Ad Blocking Recovery',
+													'google-site-kit'
+												) }
+												fullWidth
+											/>
+										</Cell>
+									</Row>
+								</Grid>
+
+								<Grid className="googlesitekit-ad-blocking-recovery__content">
+									<Row>
+										<Cell mdSize={ 6 } lgSize={ 8 }>
```

Comment:
This could also be another component for `Content` perhaps, which would essentially just be the grid, but also include the graphic so that `children` would essentially just be the `Stepper` here.

Reply:
Updated accordingly, thank you!